💄 Style makeover post release bug fixes

olyism opened the pull request:
- [x] Load the correct font file
- [x] Fix issue of webfont not loading
  - [x] Use `url` not `asset-url` for externally hosted fonts
  - [x] `preload` fonts to prevent FOUT
  - [x] Removed Circular from `_head.html.erb`
- [x] SVG diagram missing the new web font
- [x] Content fix: beta flags for test collector
- [x] Strange pink Safari toggle icon
- [x] TOC highlight fix: click event no longer overwritten by the scroll event
- [x] Consistent, slightly reduced table font sizes

olyism commented:
# Screenshots for the pink Safari icon fix

## Before (Safari) 🐽

![Screen Shot 2023-02-23 at 12 46 19 PM (2)](https://user-images.githubusercontent.com/7202667/220803951-844e8a37-6367-454e-a2b4-33b60822390c.png)

## After ✅

### Safari 🟣

![Screen Shot 2023-02-23 at 12 47 49 PM](https://user-images.githubusercontent.com/7202667/220804056-e2cb09b5-3c89-499f-8d6e-1f696010a725.png)

### Chrome 🟣

<img width="367" alt="Screen Shot 2023-02-23 at 12 48 18 PM" src="https://user-images.githubusercontent.com/7202667/220804102-5ef70454-4e1a-4e88-aac2-900a8117a515.png">

### Firefox 🟣

<img width="581" alt="Screen Shot 2023-02-23 at 12 51 20 PM" src="https://user-images.githubusercontent.com/7202667/220804502-83b4922c-cb39-4d7f-9585-fbc5fdef5843.png">

olyism commented:
Web fonts for the SVG diagram has been fixed. 

<img width="1012" alt="Screen Shot 2023-02-23 at 2 45 44 PM" src="https://user-images.githubusercontent.com/7202667/220817051-6bc9db5f-91bf-46b0-b6cb-da8ab6588ce0.png">

It's the only instance in the entire Docs site where an SVG diagram uses text.

olyism commented:
Added beta tags for beta collectors.

<img width="704" alt="Screen Shot 2023-02-23 at 2 47 10 PM" src="https://user-images.githubusercontent.com/7202667/220817183-bb47cf8e-58f5-423d-926e-02af087a7fad.png">

<img width="715" alt="Screen Shot 2023-02-23 at 2 47 26 PM" src="https://user-images.githubusercontent.com/7202667/220817219-a27b9abd-9708-42ad-9f42-52d7f42a66ef.png">

olyism commented:
Webfonts now loading correctly (as seen on Render).

<img width="766" alt="Screen Shot 2023-02-24 at 7 46 32 AM" src="https://user-images.githubusercontent.com/7202667/221026421-b41c6e41-c1ca-4908-831b-f39ec07ff132.png">

olyism commented:
hi @dannymidnight you're right, the font file isn't correct. I've downloaded the files from the Marketing Site and will need some help uploading them again. Thanks 🙏

BrettJay commented:
This might be a silly question, but I know with Rails and Next.js projects it’s possible to serve assets from a `/public` directory which ensures a stable URL without asset hashing for cache busting...

e.g. `buildkite.com/_site/fonts/aeonik.woff2`

(That doesn’t exist, an actual example of a stable public url asset hosted from site is `https://buildkite.com/_site/email/buildkite.png`)

Would that be suitable for webfonts in docs? My suspicion is that the URL we’re sourcing the web fonts here isn’t intended to be stable and so Docs would need to know when the URL changes.

dannymidnight commented on app/views/application/_head.html.erb:
```diff
@@ -1,9 +1,11 @@
-<link rel="preload" as="font" type="font/woff2" href="https://www2.buildkiteassets.com/assets/lineto-circular/lineto-circular-book-39c5d1ed54e49102939d0280aeb20f01ef021bf5ffa74dc25fcafb43fce62ff3.woff2" crossorigin="anonymous" />
-<link rel="preload" as="font" type="font/woff2" href="https://www2.buildkiteassets.com/assets/lineto-circular/lineto-circular-bold-23e71bdfef13622d0d52b2b4b7ed3c1edb9e81f210692130dee9a521e97d062f.woff2" crossorigin="anonymous" />
-<link rel="preload" as="font" type="font/woff2" href="https://www2.buildkiteassets.com/assets/lineto-circular/lineto-circular-black-a8002849ea799a4dcf1be0b1abe0f010991cbae508f24f62e9ee0262590197eb.woff2" crossorigin="anonymous" >
+<link rel="preload" as="font" type="font/woff2" href="https://www2.buildkiteassets.com/assets/aeonik/Aeonik-Bold-78fff47500f27f2f0d8d777c9d431a87036e566af7577a58ccc0776b4eb4b994.woff2" crossorigin="anonymous" />
```
Nice 👍

dannymidnight commented on app/views/application/_head.html.erb:
```diff
@@ -1,9 +1,11 @@
-<link rel="preload" as="font" type="font/woff2" href="https://www2.buildkiteassets.com/assets/lineto-circular/lineto-circular-black-a8002849ea799a4dcf1be0b1abe0f010991cbae508f24f62e9ee0262590197eb.woff2" crossorigin="anonymous" >
+<link rel="preload" as="font" type="font/woff2" href="https://www2.buildkiteassets.com/assets/aeonik/Aeonik-Bold-78fff47500f27f2f0d8d777c9d431a87036e566af7577a58ccc0776b4eb4b994.woff2" crossorigin="anonymous" />
+<link rel="preload" as="font" type="font/woff2" href="https://www2.buildkiteassets.com/assets/maison-neue/MaisonNeue-Book-5e1404c0c8bc64daaae66d3a37fa7b0c7ffe65340d181e9068741deff8669cda.woff2" crossorigin="anonymous" />
+<link rel="preload" as="font" type="font/woff2" href="https://www2.buildkiteassets.com/assets/maison-neue/MaisonNeue-BookItalic-2a5e47b91af0050b44c21f9038ef63a37e228fd53b0289dfccbcdd3aa44ba19c.woff2" crossorigin="anonymous" />
```
I think we should drop preloading these until we've landed the revised fonts. https://github.com/buildkite/buildkite/pull/11319

dannymidnight replied:
You should be able to drop the font changes after a rebase.

olyism replied:
reverting back to Maison Bold, rather than Medium, because for Docs' use case the Medium font weight is too light. It makes sense for the visual design for Docs to diverge from Marketing Site here.

olyism commented:
hi @BrettJay not a silly question at all. I struggled with serving assets from Docs' `public` folder and, to the best of my knowledge, it may have something to do with sprockets. I could load fonts locally on my local dev environment, but it was throwing 500 errors on Render 🤷‍♂️ 

In light of this, we went with the existing font approach — i.e. hosting on our buckets. This isn't ideal, it works for now, but we'd be keen to explore better solutions in the future.

I'm thinking the ideal state would be Docs loading assets from the same source as the Marketing Site in the future.

dannymidnight commented on app/assets/javascripts/toc.js:
```diff
@@ -42,6 +38,14 @@
+
+    tocLinkNodes.forEach((link) => {
+      link.addEventListener("click", (e) => {
+        setTimeout(() => setCurrentLink(e.target.hash), 25);
```
What's the thinking behind adding the timeout here?

olyism replied:
So that it highlights link that you've clicked as the active TOC item, rather than the heading in the viewport.

What's happening is both the scroll position of heading and the clicked TOC item are competing, and the scroll event always trump.

I tried a bunch of different ways and this was the best I could come up with to ensure the click comes after the scroll position with setTimeout.

dannymidnight commented on app/assets/stylesheets/base/_text_content.scss:
```diff
@@ -9,8 +9,8 @@
-  font-size: 1.1875rem;
-  line-height: 1.618;
+  font-size: $font-size-body;
+  line-height: map-get($line-heights, 'golden');
```
✨

dannymidnight commented on app/assets/stylesheets/components/_nav.scss:
```diff
@@ -228,12 +218,12 @@
     &::before {
       background: transparent asset-url("icons/angle-down.svg") 50% 50% no-repeat;
       transform: rotate(-90deg);
+      will-change: transform;
```
👍

dannymidnight reviewed (approved):
Nice one, looks like some good adjustments.

olyism commented:
hi @dannymidnight I've rebased the latest changes from `main`. Would be keen to get your eyes on it one more time to ensure before merging:

1. **Font:** loading Maison Bold, instead of Medium, because the proper bold contrasts better on Docs. We need bolding more than Marketing so it makes sense to diverge here
2. SVG diagram now loading Maison, and not Circular
3. Beta flags for beta test collectors
4. Fixed TOC highlight with setTimeout (be keen to hear if there's a more elegant solution)
5. Consistent table font size

![Screen Shot 2023-03-06 at 5 11 12 PM](https://user-images.githubusercontent.com/7202667/223032365-ff950ac6-1087-4dca-9bac-906b5cdbf7a8.png)
